libs/ngxtension/trackby-id-prop/src/trackby-id-prop.ts

```diff
@@ -0,0 +1,42 @@
+//INSPIRED BY https://medium.com/ngconf/make-trackby-easy-to-use-a3dd5f1f733b
+import { NgForOf } from '@angular/common';
+import {
+	Directive,
+	Input,
+	Provider,
+	inject,
+	type NgIterable,
+} from '@angular/core';
+
+@Directive({
+	selector: '[ngForTrackById]',
+	standalone: true,
+})
+export class NgForTrackByIdDirective<T extends { id: string | number }> {
+	@Input() ngForOf!: NgIterable<T>;
+	private ngFor = inject(NgForOf<T>, { self: true });
+
+	constructor() {
+		this.ngFor.ngForTrackBy = (index: number, item: T) => item.id;
+	}
+}
+
+@Directive({
+	selector: '[ngForTrackByProp]',
+	standalone: true,
+})
+export class NgForTrackByPropDirective<T> {
+	@Input() ngForOf!: NgIterable<T>;
+	private ngFor = inject(NgForOf<T>, { self: true });
+
+	@Input()
```

Comment:
suggestion: maybe we can use `{required: true}` here

Reply:
If I don't remember incorrectly required was added in NG16 I don't remember/checked the minimum version for peer-dep of ngxtensions?  If it's just 16+ for me no problem at all to use the required input and drop additional check in the setter